controllers/agentpool_controller_autoscaling.go

```diff
@@ -170,11 +187,6 @@ func (r *AgentPoolReconciler) reconcileAgentAutoscaling(ctx context.Context, ap
 
 	ap.log.Info("Reconcile Agent Autoscaling", "msg", "new reconciliation event")
 
-	if ap.coolDownSecondsRemaining() > 0 {
-		ap.log.Info("Reconcile Agent Autoscaling", "msg", "autoscaler is within the cooldown period, skipping")
-		return nil
-	}
-
```

Comment:
I had to move this logic further down, because here we don't have access to the current and desired number of replicas in order to determine if we are scaling up or down.